### DISCUSSION TITLE
CompositeCommand and Multiple shell

### DISCUSSION THREAD

**Discussion by GrandStrateguerre** [PrismLibrary/Prism#2765] [WPF]
Hello all,

I am able through Brian's course to operate multiple shells whith Prism.

But I've got a little problem with the composite command SaveAll which is in the shell.
When I create a new shell and I click on saveAll on this new Shell , the SaveAll Command is also fired on the first Shell.

According to Brian's Courses I've implmented all the stuff for dealing with multiple shell :
![image](https://user-images.githubusercontent.com/86523070/187154214-f0ff4409-f2f2-4830-85d2-930c67a4728d.png)
So everythings seems to works good except the CompositeCommand which is fired everywhere on each new shell instance

Do you have any suggestions to correct this problem?

Thanks.

**Comment by taso131** [comment]
Hey, I think it looks like your application shares the same instance of a module and thus it executes everywhere.
I had a similar problem and scoped regions kind of helped me with it.

Brian refered to this link which was very helpful
https://app.pluralsight.com/library/courses/prism-showing-multiple-shells/
In this course there is a scene where he executes a command in one shell and it gets executed at another open window.
This is likely to the open windows sharing the same region manager.
I think you have kind of the same problem, but with another class maybe?

It is a little bit outdated, but with a little thinking you can probably find a working solution
Good luck!
